src/Microsoft.OpenApi/Services/CopyReferences.cs

```diff
@@ -99,7 +100,8 @@ private void AddParameterToComponents(IOpenApiParameter? parameter, string? refe
     {
         EnsureComponentsExist();
         EnsureParametersExist();
-        if (Components.Parameters is not null && parameter is not null && referenceId is not null && !Components.Parameters.ContainsKey(referenceId))
+        Components.Parameters ??= new Dictionary<string, IOpenApiParameter>();
```

Comment:
I'd move the assignment within the condition, and use null prop operator in the condition, this way we don't assign if the other members of the condition are false.